tensorboard/pip_package/BUILD

```diff
@@ -5,9 +5,36 @@ package(default_visibility = ["//visibility:private"])
 
 licenses(["notice"])  # Apache 2.0
 
-# rm -rf /tmp/tensorboard
-# bazel run //tensorboard/pip_package:build_pip_package
-# pip install -U /tmp/tensorboard/*py2*.pip
+genrule(
+    name = "pip_package",
+    # Pip packages are provided in a tarball because we can't statically
+    # know the file names: they must contain the TensorBoard version,
+    # which is only defined in a Python module. Dependents of this
+    # target can extract the tarball and glob its contents.
+    outs = ["pip_packages.tar.gz"],
+    cmd = "\n".join([
+        "tmpdir=\"$$(mktemp -d)\" &&",
```

Comment:
It's hard to read this code embedded in the genrule definition like this.

What if we just made deterministic_tar_gz a data dep for build_pip_package, so that if the argument (currently named OUTPUT_DIR) ends in `.tar.gz` it emits the pip packages as an archive in the given file instead?

If we did that and also ensured that pip package building doesn't emit extraneous output, then it seems like we could make the entire genrule here just a one-liner: `$(location :build_pip_package) $@`.

Reply:
Okay; that’s totally fine with me if we just make it always emit a
`.tar.gz`, and stomachable if it cases on the filename. Will change
tomorrow.